backend/chain.go

```diff
@@ -368,6 +368,84 @@ func (g *gnoZenaoChain) UserAddress(userID string) string {
 	return gnolang.DerivePkgAddr(g.userRealmPkgPath(userID)).String()
 }
 
+// CreatePoll implements ZenaoChain
+func (g *gnoZenaoChain) CreatePoll(userID string, req *zenaov1.CreatePollRequest) error {
+	userRealmPkgPath := g.userRealmPkgPath(userID)
+	eventPkgPath := g.eventRealmPkgPath(req.EventId)
+	feedID := gnolang.DerivePkgAddr(eventPkgPath).String() + ":main"
+	options := ""
+	for _, option := range req.Options {
+		options += fmt.Sprintf(`%q, `, option)
+	}
+	kind := ""
+	if req.MultipleChoices {
+		kind = "pollsv1.POLL_KIND_MULTIPLE_CHOICE"
+	} else {
+		kind = "pollsv1.POLL_KIND_SINGLE_CHOICE"
+	}
+
+	broadcastRes, err := checkBroadcastErr(g.client.Run(gnoclient.BaseTxCfg{
+		GasFee:    "1000000ugnot",
+		GasWanted: 100000000,
+	}, vm.MsgRun{
+		Caller: g.signerInfo.GetAddress(),
+		Package: &gnovm.MemPackage{
+			Name: "main",
+			Files: []*gnovm.MemFile{{
+				Name: "main.gno",
+				Body: fmt.Sprintf(`package main
+
+import (
+	"gno.land/p/demo/ufmt"
+	"gno.land/p/zenao/daokit"
+	feedsv1 "gno.land/p/zenao/feeds/v1"
+	pollsv1 "gno.land/p/zenao/polls/v1"
+	"gno.land/r/zenao/polls"
+	"gno.land/r/zenao/social_feed"
+	user %q
+)
+
+func main() {
+	daokit.InstantExecute(user.DAO, daokit.ProposalRequest{
+		Title: "Add new poll",
+		Message: daokit.NewInstantExecuteMsg(user.DAO, daokit.ProposalRequest{
+			Title: "Add new poll",
+			Message: daokit.NewExecuteLambdaMsg(
+				NewPoll,
+			),
+		}),
+	})
+}
+
+func NewPoll() {
+	question := %q
+	options := []string{%s}
+	p := polls.NewPoll(question, %s, %d, options, nil)
+	uri := ufmt.Sprintf("/poll/%%s/gno/gno.land/r/zenao/polls", p.ID.String())
+
+	feedID := %q
+	post := &feedsv1.Post{
+		Loc:  nil,
```

Comment:
Maybe not but does some of the Zenao events have a location ? maybe we can set it on the post :) if not just ignore this

Reply:
for now, the post location is {lat, lng} but the location of an event can be an uri toward a google meet or be a custom address. i think we can ignore this for now and think about it later
I think the best option would be to allow people to choose location of the post through a map picker for example